File: leetcode-cn/sword2offer2/000剑指0_Offer_II_074._合并区间.java
/*
执行用时：10 ms, 在所有 Java 提交中击败了7.44% 的用户
内存消耗：41.3 MB, 在所有 Java 提交中击败了5.21% 的用户
通过测试用例：168 / 168
*/

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        List<int[]> ans = new ArrayList<>();
        for(int[] interval: intervals) {
            int left = interval[0], right = interval[1];
            if (ans.size() == 0 || ans.get(ans.size() - 1)[1] < left)
                ans.add(new int[]{left, right});
            else
                ans.get(ans.size() - 1)[1] = Math.max(ans.get(ans.size() - 1)[1], right);
        }

        return ans.toArray(new int[ans.size()][]);
    }
}

File: leetcode-cn/sword2offer2/000剑指0_Offer_II_074._合并区间.py
'''
执行用时：36 ms, 在所有 Python3 提交中击败了77.78% 的用户
内存消耗：15.5 MB, 在所有 Python3 提交中击败了89.58% 的用户
通过测试用例：168 / 168
'''

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        ans = []
        for interval in intervals:
            start, end = interval
            if not ans or ans[-1][1] < start:
                ans.append(interval)
            elif ans[-1][1] < end:
                ans[-1][1] = end
        return ans

'''
输入：
[[1,4],[2,3]]
输出：
[[1,3]]
预期结果：
[[1,4]]
'''

